src/components/QuickSearch/util.tsx

```diff
@@ -0,0 +1,60 @@
+import { isDefined } from '~/utils/type-guards';
+
+type MatchedFilter = {
+  indexName: string;
+  attribute: string;
+  attributeRegexp: RegExp;
+  matchRegexp: RegExp;
+  matches: string[] | null;
+};
```

Comment:
This will be replaced with the following
```suggestion
const filters = [
  {
    indexName: 'models',
    attribute: 'type',
    attributeRegexp: /t:(\w+)/,
    matchRegexp: /(?<=t:)\w+/,
    matches: [],
  },
] as const;

type MatchedFilter = (typeof filters)[number];
type FilterIndex = (typeof filters)[number]['indexName'];
```

Reply:
Good idea, but if we go that route we won't be able to set "matches" as a manipulable/nullable array (which is it by definition).

Reply:
Ahhh right, then what about
```suggestion
type FilterIndex = 'models' | 'users' | 'tags' | 'articles';
type MatchedFilter = {
  indexName: FilterIndex;
  attribute: string;
  attributeRegexp: RegExp;
  matchRegexp: RegExp;
  matches: string[] | null;
};

const filters: MatchedFilter[] = [
  {
    indexName: 'models',
    attribute: 'type',
    attributeRegexp: /t:(\w+)/,
    matchRegexp: /(?<=t:)\w+/,
    matches: [],
  },
];
```